### PR TITLE
[Release Test] Fix dask on ray 1tb sort failure.

### LINKED ISSUE #36812: Release test dask_on_ray_1tb_sort failed
Release test **dask_on_ray_1tb_sort** failed. See https://buildkite.com/ray-project/release-tests-branch/builds/1828#0188f85c-1630-4e2a-83c4-cbd288f97aed for more details.

Managed by OSS Test Policy

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

Looks like some of s3fs versions cause this failures. Installing the latest version seems to fix the issue. 

## Related issue number

Fixes https://github.com/ray-project/ray/issues/36812

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by jjyao — APPROVED** [review]

File: release/nightly_tests/dask_on_ray/dask_on_ray_app_config.yaml [hunk 1]
```diff
@@ -13,6 +13,7 @@ python:
 post_build_cmds:
   # - pip install fastparquet
   - pip3 install boto3 s3fs
+  - pip3 install -U s3fs
```
Comment (can-anyscale) [inline release/nightly_tests/dask_on_ray/dask_on_ray_app_config.yaml:16]: Can you add to this file as well https://github.com/ray-project/ray/blob/master/release/ray_release/byod/byod_dask_test.sh#L9? This is for the BYOD version of the test. Thankkks

**Comment by rkooo567** [comment]
The tests pass. I will merge it after addressing @can-anyscale 's comment

**Review by aslonnie — COMMENTED** [review]
(leaving this for @can-anyscale to review)

**Review by can-anyscale — APPROVED** [review]

**Comment by can-anyscale** [comment]
Oh, didn't know I have to stamp. Go for it!

**Pull request merged**